netty/src/test/java/io/grpc/netty/UdsNettyChannelProviderTest.java

```diff
@@ -0,0 +1,198 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.netty;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import io.grpc.Grpc;
+import io.grpc.InsecureChannelCredentials;
+import io.grpc.InternalServiceProviders;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.ManagedChannelProvider;
+import io.grpc.ManagedChannelProvider.NewChannelBuilderResult;
+import io.grpc.ManagedChannelRegistryAccessor;
+import io.grpc.TlsChannelCredentials;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import io.grpc.testing.protobuf.SimpleRequest;
+import io.grpc.testing.protobuf.SimpleResponse;
+import io.grpc.testing.protobuf.SimpleServiceGrpc;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerDomainSocketChannel;
+import io.netty.channel.unix.DomainSocketAddress;
+import java.io.IOException;
+import org.junit.After;
+import org.junit.Assume;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link UdsNettyChannelProvider}. */
+@RunWith(JUnit4.class)
+public class UdsNettyChannelProviderTest {
+
+  @Rule public TemporaryFolder tempFolder = new TemporaryFolder();
+
+  @Rule
+  public final GrpcCleanupRule cleanupRule = new GrpcCleanupRule();
+
+
+  private UdsNettyChannelProvider provider = new UdsNettyChannelProvider();
+
+  private EventLoopGroup elg;
+  private EventLoopGroup boss;
+
+  @After
+  public void tearDown() {
+    if (elg != null) {
+      elg.shutdownGracefully();
+    }
+    if (boss != null) {
+      boss.shutdownGracefully();
+    }
+  }
+
+  @Test
+  public void provided() {
+    for (ManagedChannelProvider current
+        : InternalServiceProviders.getCandidatesViaServiceLoader(
+            ManagedChannelProvider.class, getClass().getClassLoader())) {
+      if (current instanceof UdsNettyChannelProvider) {
+        return;
+      }
+    }
+    fail("ServiceLoader unable to load UdsNettyChannelProvider");
+  }
+
+  @Test
+  public void providedHardCoded() {
+    for (Class<?> current : ManagedChannelRegistryAccessor.getHardCodedClasses()) {
+      if (current == UdsNettyChannelProvider.class) {
+        return;
+      }
+    }
+    fail("Hard coded unable to load UdsNettyChannelProvider");
+  }
+
+  @Test
+  public void basicMethods() {
+    assertTrue(provider.isAvailable());
```

Comment:
You meant assumeTrue.

Reply:
Yes, with the latest change made to `UdsNettyChannelProvider` it should be `assumeTrue`. Otherwise this was copied from `NettyChannelProviderTest` which used `assertTrue` . Good catch. Making the change